spec/lib/unique_within_region_validator_spec.rb

```diff
@@ -84,28 +65,48 @@
 
       it "is case insensitive if match_case is set to false" do
         in_first_region_rec      = case_insensitive_class.find(in_first_region_id)
-        also_in_first_region_rec = case_insensitive_class.find(also_in_first_region_id)
+        also_in_first_region_rec = case_insensitive_class.new(:id => case_sensitive_class.id_in_region(2, 0), :name => test_name)
         in_second_region_rec     = case_insensitive_class.find(in_second_region_id)
 
-        expect(in_first_region_rec.valid?).to be false
+        expect(in_first_region_rec.valid?).to be true
         expect(also_in_first_region_rec.valid?).to be false
         expect(in_second_region_rec.valid?).to be true
       end
 
       it "applies the passed scope" do
         in_first_region_rec           = scoped_class.find(in_first_region_id)
         also_in_first_region_rec      = scoped_class.find(also_in_first_region_id)
-        new_email_in_first_region_rec = scoped_class.find(new_email_in_first_region_id)
         in_second_region_rec          = scoped_class.find(in_second_region_id)
 
         expect(in_first_region_rec.valid?).to be true
         expect(also_in_first_region_rec.valid?).to be true
-        expect(new_email_in_first_region_rec.valid?).to be true
         expect(in_second_region_rec.valid?).to be true
 
-        new_email_in_first_region_rec.email = test_email
+        also_in_first_region_rec.name = test_name
+
+        expect(also_in_first_region_rec.valid?).to be false
+      end
+
+      it "gracefully handles no name" do
+        rec = case_sensitive_class.new
+        expect { rec.valid? }.not_to make_database_queries
```

Comment:
These `.valid?` checks verify a database query or not, which is good, but don't actually check that `valid?` returns the correct value.  Do we have tests for the correct value being returned for the below cases (new, new with name, etc.)?

Reply:
@Fryguy these are tested elsewhere but...
I added expects here - no real penalty to be positive that these values work